src/client/stdlib/duckdb.js

```diff
@@ -254,6 +254,14 @@ async function insertFile(database, name, file, options) {
         if (/\.parquet$/i.test(file.name)) {
           return await connection.query(`CREATE VIEW '${name}' AS SELECT * FROM parquet_scan('${file.name}')`);
         }
+        if (/\.(db|ddb|duckdb)$/i.test(file.name)) {
+          // https://duckdb.org/internals/storage.html#storage-header
+          const magic = await fetch(await file.url(), {headers: {range: "bytes=0-11"}})
+            .then((d) => d.blob())
+            .then((d) => d.slice(8, 12).text());
+          if (magic === "DUCK") return await connection.query(`ATTACH '${file.name}' AS ${name} (READ_ONLY)`);
+          else console.warn(`${file.name} does not appear to be a valid DuckDB database file.`);
```

Comment:
Should we just pass any file through and let `connection.query` error as needed?

Reply:
Yes we should probably do that—maybe it will bork the DuckDB instance in some cases, and the user will have to reload the page, but it will be simpler (for us) and a better separation of concerns. All the cases we want to support for now work in both cases.

In the future, we could extend the sql front matter syntax to pass a type, esp. for remote files:

```yaml
sql:
  gaia:
    source: "an opaque external url"
    type: text/csv
```

More generally in this file, the code branches based on file.name or file.mimeType, depending on something that isn't specified: is it because the remote file could be served with a wrong file name and a correct mime type in some cases? It's something we'll have to clarify.

But for now, yeah, we can do the simplest. The native DuckDB errors look like this:
* `Error: Invalid Error: Opening file 'base.udb' failed with error: Failed to open file: unknown.db`
* `Error: IO Error: The file "index.md" exists, but it is not a valid DuckDB database file!`

I think they're quite expressive and understandable!